### PR TITLE
Restructuring

### PR DESCRIPTION
**Pull request by LimbeckKat**
Based on what we discussed last week I changed the structure of the package, split up the functionalities some more, and added more documentation to make the code easier to understand.

I've also added a class, `Diversipy`, that computes magnitude functions and the diversity measures across multiple spaces and does the automated scale-finding across multiple spaces as detailed in the NeurIPS paper. 

Ready to merge? Thank you for having a look!

### CODE REVIEW

File: magnipy/magnitude/dimension.py
Comment (jeremy-wayland) [inline magnipy/magnitude/dimension.py:1]: I think a naming convention here might be nice (although not necessary!): having files in `compute.py` named `compute_*` and then _not_ using the compute prefix in other functions.
E.g.:
* `magnipy.magnitude.compute.magnitude_from_distances` ->  `magnipy.magnitude.compute.compute_magnitude_from_distances`
* `mangnipy.magnitude.dimension.compute_magnitude_dimension` -> `mangnipy.magnitude.dimension.get_magnitude_dimension`

Maybe @emsimons is already addressing this in #1 :)

**Review by jeremy-wayland — CHANGES_REQUESTED** [review]
Nice work!! Restructuring looks great🙌

Requesting some changes because I don't think the tests were updated to suit new structure– both `test_graphs` and `test_mag` are failing.

 ## test_mag
Easy fix here, there is still a legacy `inv` method included in the supported methods. I assume this is now `pinv`? But theres no check here in the following code block for `inv`

``` python
else:
        if method == "scipy":
            mag_fn = weights_scipy
        elif method == "scipy_sym":
            mag_fn = weights_scipy_sym
        elif method == "cholesky":
            mag_fn = weights_cholesky
        elif method == "conjugate_gradient_iteration":
            mag_fn = weights_cg
        elif method == "pinv":
            mag_fn = weights_pinv
        elif method == "naive":
            mag_fn = weights_naive
        else:
            raise Exception(
                "The computation method must be one of 'cholesky', 'scipy', 'scipy_sym', 'inv', 'pinv', 'conjugate_gradient_iteration', 'cg', 'naive', 'spread'."
            )
```


Since theres no matching weight_fn `weights_inv` I assume this just needs to be removed from supported methods and then tests!


## test_graphs
Also need to remove `inv`. Main issue is `spread` is not getting the proper analytic solution... I assume this is expected but not totally sure– wil leave this to the Magnitude expert @LimbeckKat😊
Removing spread makes these tests pass!

File: magnipy/magnitude/dimension.py
Comment (LimbeckKat) [reply magnipy/magnitude/dimension.py:1]: Good point! I'll keep this in mind and have adjusted some of the names.

**Comment by LimbeckKat** [comment]
Well spotted! I have fixed the tests and also added analytic tests for computing spread.

**Review by jeremy-wayland — APPROVED** [review]
Looks great and all tests passing locally✅🚀

Ready to merge!!

**Pull request merged**